v2/network/session.go

```diff
@@ -275,6 +276,13 @@ func (session *Session) negotiate() {
 	if !connOption.SSLVerify {
 		config.InsecureSkipVerify = true
 	}
+
+	if tlsConfig := session.Context.ConnOption.TLSConfig; tlsConfig != nil {
```

Comment:
should this overwrite go to the top of the `negotiate` to skip setting up the default `config`?

Reply:
I have moved this at the top so the default config will be unchanged.